terraform/main.tf

```diff
@@ -1,7 +1,31 @@
 # Copyright (c) 2023, Oracle and/or its affiliates.
 # Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
 
+module "network-validation" {
+  source = "./modules/network-validator"
+  count  = local.use_existing_subnets && !var.skip_network_validation ? 1 : 0
+  wls_subnet_id = var.wls_subnet_id
+  bastion_subnet_id = var.is_bastion_instance_required ? var.bastion_subnet_id : ""
+  bastion_ip        = var.is_bastion_instance_required && var.existing_bastion_instance_id != "" ? data.oci_core_instance.existing_bastion_instance[0].private_ip : ""
+  lb_subnet_1_id    = var.add_load_balancer ? var.lb_subnet_1_id : ""
+  lb_subnet_2_id    = var.add_load_balancer && !local.use_regional_subnet ? var.lb_subnet_2_id : ""
+  mount_target_subnet_id    = var.add_fss ? var.mount_target_subnet_id : ""
+  atp_db_id =  !local.is_oci_db ? var.atp_db_id : ""
+  oci_db_dbsystem_id = local.is_oci_db ? var.oci_db_dbsystem_id : ""
+  oci_db_port = local.is_oci_db ? var.oci_db_port : 0
+  db_vcn_lpg_id = local.is_oci_db ? var.db_vcn_lpg_id : ""
```

Comment:
this module will run when using existing subnets. But db vcn lpg id is needed only when using vcn peering, which is only used when existing subnets are not used. So, strictly speaking, this variable will always be empty.  So. I am not sure if it is worth having it

Reply:
Fixed it.